server/events/vcs/github_client.go

```diff
@@ -430,16 +481,106 @@ func (g *GithubClient) UpdateStatus(ctx context.Context, request types.UpdateSta
 }
 
 // [WENGINES-4643] TODO: Move the checks implementation to UpdateStatus once github checks is stable
-// UpdateChecksStatus updates the status check
 func (g *GithubClient) UpdateChecksStatus(ctx context.Context, request types.UpdateStatusRequest) error {
-	// TODO: Implement updating github checks
-	// - Get all checkruns for this SHA
-	// - Match the UpdateReqIdentifier with the check run. If it exists, update the checkrun. If it does not, create a new check run.
+	result, _, err := g.client.Checks.ListCheckRunsForRef(ctx, request.Repo.Owner, request.Repo.Name, request.Ref, &github.ListCheckRunsOptions{})
```

Comment:
Shouldn't this be using `GetRepoChecks`?  Or at least you can refactor that into private method to so both methods can use it.

Reply:
Yup, I refactored it to `GetRepoChecksForRef`. I can't make it private since it's being used in the instrumented_client.go file.